js/modules/k6/http/response.go

```diff
@@ -51,11 +52,28 @@ func responseFromHttpext(resp *httpext.Response) *Response {
 func (res *Response) JSON(selector ...string) goja.Value {
 	v, err := ((*httpext.Response)(res)).JSON(selector...)
 	if err != nil {
-		common.Throw(common.GetRuntime(res.GetCtx()), err)
+		var value interface{}
+		var body []byte
+		switch b := res.Body.(type) {
+		case []byte:
+			body = b
+		case string:
+			body = []byte(b)
+		}
+		if err := json.Unmarshal(body, &value); err != nil {
+			switch t := err.(type) {
+			case *json.SyntaxError:
+				err = checkErrorInJSON(body, int(t.Offset), err)
+			default:
+				break
+			}
+			common.Throw(common.GetRuntime(res.GetCtx()), err)
+		}
```

Comment:
As you have seen we have moved practically all of the JSON code to `httpext` I would prefer if this new code is moved there as well